### PR TITLE
Removal of ASD node ID environment variable

### PR DESCRIPTION
**Pull request by kvanhijf**

### CODE REVIEW

File: source/asdmanager.py [hunk 1]
```diff
@@ -247,42 +233,43 @@ def _validate_and_retrieve_pre_config():
...
     if 'asdmanager' not in config or not isinstance(config['asdmanager'], dict):
-        print Interactive.boxed_message(['The ASD manager pre-configuration file must contain a "asdmanager" key with a dictionary as value'])
+        print '\n' + Interactive.boxed_message(['The ASD manager pre-configuration file must contain a "asdmanager" key with a dictionary as value'])
         sys.exit(1)
 
     errors = []
     config = config['asdmanager']
     actual_keys = config.keys()
-    expected_keys = ['api_ip', 'api_port', 'asd_ips', 'asd_start_port']
+    expected_keys = ['api_ip', 'api_port', 'asd_ips', 'asd_start_port', 'configuration_store']
```
Comment (JeffreyDevloo) [inline source/asdmanager.py]: Configuration store is now a must? Or is it an allowed param :thinking:

File: source/asdmanager.py [hunk 2]
```diff
@@ -247,42 +233,43 @@ def _validate_and_retrieve_pre_config():
...
 
     try:
         ExtensionsToolbox.verify_required_params(actual_params=config,
                                                  required_params={'api_ip': (str, ExtensionsToolbox.regex_ip, True),
                                                                   'asd_ips': (list, ExtensionsToolbox.regex_ip, False),
                                                                   'api_port': (int, {'min': 1025, 'max': 65535}, False),
-                                                                  'asd_start_port': (int, {'min': 1025, 'max': 65435}, False)})
+                                                                  'asd_start_port': (int, {'min': 1025, 'max': 65435}, False),
+                                                                  'configuration_store': (str, ['arakoon'], False)})
     except RuntimeError as rte:
```
Comment (JeffreyDevloo) [inline source/asdmanager.py]: Could be handy to log this exception somewhere as well.( For automation purposes?)

File: source/controllers/update.py [hunk 3]
```diff
@@ -156,3 +161,46 @@ def restart_services():
                 service_manager.restart_service(service_name, SDMUpdateController._local_client)
             except CalledProcessError as cpe:
                 SDMUpdateController._logger.debug('Failed to restart service {0} {1}'.format(service_name, cpe))
```
Comment (JeffreyDevloo) [inline source/controllers/update.py]: Default log level was now info. RIP monitoring?

File: source/tools/configuration.py [hunk 4]
```diff
@@ -43,52 +48,47 @@ def __init__(self):
...
-                                       'ip': config['api_ip'],
-                                       'port': config['api_port'],
-                                       'version': 0},
-                       'config/network': {'ips': config['asd_ips'],
-                                          'port': config['asd_start_port']}}
-        for key, value in base_config.iteritems():
-            cls.set(key='/ovs/alba/asdnodes/{0}/{1}'.format(node_id, key),
-                    value=value,
-                    raw=False)
+        with open('/etc/openvstorage_sdm_id', 'r') as the_file:
```
Comment (JeffreyDevloo) [inline source/tools/configuration.py]: the_file does not really reflect a file with an id in it

File: source/tools/configuration.py [hunk 5]
```diff
@@ -43,52 +48,47 @@ def __init__(self):
...
+            node_id = the_file.read().strip()
+            cls.set(key=Configuration.ASD_NODE_CONFIG_MAIN_LOCATION.format(node_id),
```
Comment (JeffreyDevloo) [inline source/tools/configuration.py]: No need to have this indented under the open file as the only thing that had to be done was reading the id

**Comment by kvanhijf** [comment]
https://github.com/openvstorage/framework/pull/1750 --> openvstorage-2.10.0-dev.1504773627.f41d9d2
https://github.com/openvstorage/iscsi-manager/pull/21 --> iscsi-manager-0.2.0-dev.1504773662.65998c2
https://github.com/openvstorage/alba-asdmanager/pull/274 --> openvstorage-sdm-1.10.0-dev.1504773683.e0b11db
https://github.com/openvstorage/framework-extensions/pull/24 --> openvstorage-extensions-0.2.0-dev.1504773704.a80aea4
https://github.com/openvstorage/framework-alba-plugin/pull/463 --> openvstorage-backend-1.10.0-dev.1504773728.0e75ea4

**Pull request merged**